base/string_utils.cpp

```diff
@@ -328,4 +328,25 @@ bool AlmostEqual(string const & str1, string const & str2, size_t mismatchedCoun
   return false;
 }
 
+bool ParseCSVRow(string const & s, vector<string> & target, char const delimiter, size_t const columns)
+{
+  target.clear();
+  using It = TokenizeIterator<SimpleDelimiter, string::const_iterator, true>;
+  for (It it(s, SimpleDelimiter(delimiter)); it; ++it)
+  {
+    string column = *it;
+    strings::Trim(column);
+    target.push_back(column);
+  }
+
+  // Special case: if the string is empty, return an empty array instead of {""}.
+  if (target.size() == 1 && target[0].empty())
+  {
+    target.clear();
+    return false;
+  }
+
+  return columns <= 0 || target.size() == columns;
```

Comment:
The return value indicates whether it's a correct CSV line. If it's `true`, the caller can process it, otherwise it should ship the line. Zero is the default value for `columns`, if it is not overridden, the check is skipped, we don't compare number of columns with anything.

The empty row case is dealt with above, this check is for comparing a positive number of columns to the expected value, if there is any.